proposals/p2006.md

```diff
@@ -324,6 +396,205 @@ type system but addresses the syntactic differences separately and uses
+alternative -- this would remove the explicit visual marker for where non-local
+memory is accessed and potentially mutated.
+
+### Alternative pointer syntaxes
+
+The syntax both for declaring a pointer type and dereferencing a pointer has
+been extensively discussed in the leads question [#523].
+
+[#523]: https://github.com/carbon-language/carbon-lang/issues/523
+
+The primary sources of concern over a C++-based syntax:
+
+1.  A prefix dereference operator composes poorly with postfix and infix
+    operations.
+
+    -   This is highlighted by the use of `->` for member access due to the poor
+        composition with `.`: `(*pointer).member`.
+
+2.  Even without replicating the ["inside out"][inside-out] C/C++ challenges, we
+    would end up with a prefix, postfix, and infix operator `*`.
+
+[inside-out]:
+    https://faculty.cs.niu.edu/~mcmahon/CS241/Notes/reading_declarations.html
+
+The second issue was resolved in [#520], giving us at least the flexibility to
+consider `*` both for dereference and pointer type, but we still considered
+numerous alternatives given the first concerns. These were discussed in detail
+in a [document][pointer-syntax-doc] but the key syntax alternatives are
+extracted with modern syntax below.
+
+[pointer-syntax-doc]:
+    https://docs.google.com/document/d/1gsP74fLykZBCWZKQua9VP0GnQcADCkn5-TIC1JbUvdA/edit?resourcekey=0-8MsUybUvHDCuejrzadrbMg
+
+#### Pointer type alternative syntaxes
+
+**Postfix `*`:**
+
+```
+var p: i32* = &i;
+```
+
+Advantages:
+
+-   Most familiar to C/C++ programmers
+    -   Doesn't collide with prefix `*` for dereference so can have that
+        familiarity as well.
+
+Disadvantages:
+
+-   Requires the [#520] logic to resolve parsing ambiguities.
+-   Visually, `*` is used for both pointers and multiplication.
+-   Interacts poorly with prefix array type syntaxes.
+-   Looks like a C++ pointer but likely has different semantics.
+
+**Prefix `*`:**
+
+```
+var p: *i32 = &i;
+```
+
+Advantages:
+
+-   Used by Rust, Go, Zig,
+    [Jai](https://github.com/BSVino/JaiPrimer/blob/master/JaiPrimer.md#memory-management).
+-   Go in particular has good success transitioning C/C++ programmers to this
+    syntax.
+-   Allows reading types left-to-right.
+
+Disadvantages:
+
+-   "Pointer to T" and "Dereference T" would not be distinguished grammatically,
+    could particularly be a problem in template code
+-   `*` used for both pointers and multiplication
+-   Uncanny valley -- it is very close to C++'s syntax but not quite the same.
+
+**Prefix `&`:**
+
+```
+var p: &i32 = &i;
+```
+
+Advantages:
+
+-   "Type of a pointer is a pointer to the type"
+-   Allows reading types left-to-right.
+
+Disadvantages:
+
+-   Visual ambiguity:
+
+    ```
+    let X:! type = i32;
+    // Can't actually write this, but there is a visual collision between
+    // whether this is the address of `X` or pointer-to-`i32`.
+    var y: auto = &X;
+    ```
+
```

Comment:
Looks like something about this is confusing prettier and github's syntax highlighting.